src/outputgeneration/ParseTreeMapWriter.js

```diff
@@ -115,7 +115,8 @@ export class ParseTreeMapWriter extends ParseTreeWriter {
       column: position.column || 0  // source map uses zero based columns
     };
     if (position.source.name !== this.sourceName_) {
-      this.sourceName_ = relativeToSourceRoot(position.source.name,
+      this.sourceName_ = position.source.name;
+      this.relativeSourceName_ = relativeToSourceRoot(position.source.name,
```

Comment:
yes, and `this.sourceName_` as well, though in this case we have almost singleton compared to the work to build the sourcemap.